src/sentry/search/events/fields.py

```diff
@@ -1791,22 +1821,28 @@ def is_accessible(self, acl=None):
             ],
             default_result_type="number",
         ),
-        # Currently only used by trace meta so we can count event types which is why this only accepts strings
+        # The calculated arg will cast the string value according to the value in the column
         Function(
             "count_if",
             required_args=[
-                ColumnNoLookup("column", allowed_columns=["event.type", "http.status_code"]),
+                FunctionArg("column"),
                 ConditionArg("condition"),
                 StringArg("value"),
             ],
+            calculated_args=[
+                {
+                    "name": "typed_value",
+                    "fn": calculate_count_if_value,
```

Comment:
nit: can we name this function to something like "normalize_count_if_value" or "type_count_if_value" since it isn't actually calculating anything